src/Search/Item_Filter.php

```diff
@@ -0,0 +1,187 @@
+            return false;
+        }
+
+        // Load saved filters
+        $filter = self::getForItem($item);
+        $itemtype = $item->getItemtypeToFilter();
+        $criteria = $filter ? $filter->fields['search_criteria'] : self::getDefaultSearch($itemtype);
+        $can_edit = $item->canUpdateItem();
+
+        // Force criteria into session
+        // Even when specifying the criteria in $params, they are not taken into
+        // account and GLPI relies on session data instead
+        $_SESSION['glpisearch'][$itemtype]['criteria'] = $criteria;
+
+        // Set search engine parameters
+        $params = [
+            'criteria'                => $criteria,
+            'metacriteria'            => [],
+            'push_history'            => false,
+            'hide_controls'           => true,
+            'showmassiveactions'      => false,
+            'showbookmark'            => false,
+            'showreset'               => false,
+            'actionvalue'             => __("Preview results"),
+            'extra_actions_templates' => [
+                "components/search/items_filter_actions.html.twig" => [
+                    'itemtype'    => $item->getType(),
+                    'items_id'    => $item->getID(),
+                    'show_save'   => $can_edit,
+                    'show_delete' => $can_edit && !is_null($filter),
+                ],
+            ],
+        ];
+
+        // Print page
+        $twig = TemplateRenderer::getInstance();
+        $twig->display("components/search/item_filter.html.twig", [
+            'info_title'       => $item->getInfoTitle(),
+            'info_description' => $item->getInfoDescription(),
+            'itemtype'         => $itemtype,
+            'params'           => $params,
+            'filter_enabled'   => $filter !== null,
+        ]);
+
+        return true;
+    }
+
+    public function post_getFromDB()
+    {
+        parent::post_getFromDB();
+
+        // Decode filter
+        $this->fields['search_criteria'] = json_decode(
+            $this->fields['search_criteria'],
+            JSON_OBJECT_AS_ARRAY
+        );
+    }
+
+    public function getHistoryChangeWhenUpdateField($field)
+    {
+        // Don't show json changes in history
+        if ($field == "search_criteria") {
+            return ['0', '', ''];
+        }
+        return parent::getHistoryChangeWhenUpdateField($field);
+    }
+
+    /**
+     * Get filter for a given item
+     *
+     * @param CommonDBTM&FilterableInterface $item Given item
+     *
+     * @return null|self Null if no filter are defined for the given item
+     */
+    public static function getForItem(CommonDBTM&FilterableInterface $item): ?self
+    {
+        $filter = new self();
+        $filter_exist = $filter->getFromDBByCrit([
+            'itemtype' => $item->getType(),
+            'items_id' => $item->getID(),
+        ]);
+
+        return $filter_exist ? $filter : null;
+    }
```

Comment:
This method is only used in `FilterableTrait`. It could be moved there, made private and non static.

Reply:
Its also used in `Item_Filter`: 

![image](https://user-images.githubusercontent.com/42734840/233398933-d68de3f8-6ee5-48a2-8f67-31bae0676abe.png)

![image](https://user-images.githubusercontent.com/42734840/233399208-f6f6d686-479b-4725-8dc9-3c2c750663af.png)

I guess it could still be moved into the trait and called from $item (but obviously as a public method not private) but I'm not too sure as its not really related to the filter feature, it's just some kind of CommonDBChild boilerplate.